entities/buildinfo.go

```diff
@@ -190,13 +191,57 @@ type Module struct {
 	*Checksum
 }
 
+func (m *Module) isEqual(b Module) bool {
+	return isEqualArtifactSlices(m.Artifacts, m.Artifacts) && isEqualDependencySlices(m.Dependencies, b.Dependencies) && m.Id == b.Id && m.Type == b.Type
```

Comment:
Looks like a bug - 
```
isEqualArtifactSlices(m.Artifacts, m.Artifacts) 
```

Also, for efficiency reasons, better to move the "cheap checks" to the beginning of the if statement -  
```
m.Id == b.Id && m.Type == b.Type
```